src/Data/List/Relation/Unary/All/Properties.agda

```diff
@@ -444,15 +444,21 @@ drop⁺ (suc n) (px ∷ pxs) = drop⁺ n pxs
 
 dropWhile⁺ : (Q? : Decidable Q) → All P xs → All P (dropWhile Q? xs)
```

Comment:
I can't quite see what a judicious generalisation of this should be from which this would follow by analogy with `takeWhile⁺` below!?

Reply:
Should this be a corollary of something about `All P xs` implies `All P ys` whenever ys is a subset of xs?

Reply:
Indeed. 

As with #2509 / #2513 I'm worried that this needs much more careful consideration of the complicated interlocking dependencies between `All`/`Any` and `Sublist`/`Subset`, and in this case, I'd been after something closer to the `takeWhileP⇒Q⇒R⁺` introduced by this PR.

The 'problem' is that `take`/`drop` are really concerned with `Prefix`/`Suffix`... so perhaps a larger scale refactoring might be necessary later?

Reply:
Well, `take` / `drop` don't really respect `Prefix` / `Suffix`, do they? `takeWhile` and `dropWhile` do. And so for them, it does seem to make sense to think more deeply about what the whole structure should be.

Reply:
Hmm. Both `take`/`drop` (provided the argument `n` is within bounds...;-)) and `takeWhile`/`dropWhile` enjoy a characteristic property that their append yields the original list?
But agree: more thought required, or else actually abandoning this PR!

Reply:
> Should this be a corollary of something about `All P xs` implies `All P ys` whenever ys is a subset of xs?

Indeed, and this appears *twice* in the library:
* `Data.List.Relation.Unary.All.Properties.anti-mono : xs Subset.⊆ ys → All P ys → All P xs` (and `Sublist` is a stronger property than `Subset`)
* `Data.List.Relation.Binary.Sublist.Propositional.Properties.All-resp-⊆ : {P : Pred A ℓ} → (All P) Respects _⊇_` and this should be refactored to a proof in `...Sublist.Setoid.Properties`, under assumptions that `P` respects the underlying `Setoid` equality?

Reply:
Followups on #2525 ...